win32/src/threadpool_win32.c

```diff
@@ -213,81 +201,19 @@ int threadpool_open(THANDLE(THREADPOOL) threadpool)
-            LogError("Open called in state %" PRI_MU_ENUM "", MU_ENUM_VALUE(THREADPOOL_WIN32_STATE, current_state));
-            result = MU_FAILURE;
-        }
-        else
-        {
-            /* Codes_SRS_THREADPOOL_WIN32_01_026: [ threadpool_open shall initialize a thread pool environment by calling InitializeThreadpoolEnvironment. ]*/
-            InitializeThreadpoolEnvironment(&threadpool_ptr->tp_environment);
-
-            /* Codes_SRS_THREADPOOL_WIN32_01_027: [ threadpool_open shall set the thread pool for the environment to the pool obtained from the execution engine by calling SetThreadpoolCallbackPool. ]*/
-            SetThreadpoolCallbackPool(&threadpool_ptr->tp_environment, threadpool->pool);
-
-            /* Codes_SRS_THREADPOOL_WIN32_01_028: [ threadpool_open shall create a threadpool cleanup group by calling CreateThreadpoolCleanupGroup. ]*/
-            threadpool_ptr->tp_cleanup_group = CreateThreadpoolCleanupGroup();
-            if (threadpool_ptr->tp_cleanup_group == NULL)
-            {
-                /* Codes_SRS_THREADPOOL_WIN32_01_040: [ If any error occurrs, threadpool_open shall fail and return a non-zero value. ]*/
-                LogLastError("CreateThreadpoolCleanupGroup failed");
-                result = MU_FAILURE;
-            }
-            else
-            {
-                /* Codes_SRS_THREADPOOL_WIN32_01_029: [ threadpool_open shall associate the cleanup group with the just created environment by calling SetThreadpoolCallbackCleanupGroup. ]*/
-                SetThreadpoolCallbackCleanupGroup(&threadpool_ptr->tp_environment, threadpool_ptr->tp_cleanup_group, on_io_cancelled);
-
-                /* Codes_SRS_THREADPOOL_WIN32_01_015: [ threadpool_open shall set the state to OPEN. ]*/
-                (void)InterlockedExchange(&threadpool_ptr->state, THREADPOOL_WIN32_STATE_OPEN);
-                WakeByAddressSingle((PVOID)&threadpool_ptr->state);
-
-                /* Codes_SRS_THREADPOOL_WIN32_01_012: [ On success, threadpool_open shall return 0. ]*/
-                result = 0;
-
-                goto all_ok;
-            }
-
-            DestroyThreadpoolEnvironment(&threadpool_ptr->tp_environment);
-
-            (void)InterlockedExchange(&threadpool_ptr->state, THREADPOOL_WIN32_STATE_CLOSED);
-            WakeByAddressSingle((PVOID)&threadpool_ptr->state);
-        }
+        /* Codes_SRS_THREADPOOL_WIN32_01_012: [ On success, threadpool_open shall return 0. ]*/
+        result = 0;
     }
-
-all_ok:
     return result;
 }
 
 void threadpool_close(THANDLE(THREADPOOL) threadpool)
 {
     if (threadpool == NULL)
     {
-        /* Codes_SRS_THREADPOOL_WIN32_01_016: [ If threadpool is NULL, threadpool_close shall return. ]*/
+        /* Codes_SRS_THREADPOOL_WIN32_05_018: [ If threadpool is NULL, threadpool_close shall return. ]*/
         LogError("Invalid arguments: THREADPOOL_HANDLE threadpool=%p", threadpool);
     }

```

Comment:
>} [](http://example.com/codeflow?start=4&length=1)

Consider adding an else branch with a comment of "do_nothing"

Reply:
Done